xrpl/utils/txn_parser/utils/order_book_parser.py

```diff
@@ -0,0 +1,217 @@
+"""Helper functions for order book parser."""
+
+from decimal import Decimal
+from typing import Any, Dict, List, Optional, Union
+
+from typing_extensions import Literal
+
+from xrpl.models import TransactionMetadata
+from xrpl.utils.time_conversions import ripple_time_to_posix
+from xrpl.utils.txn_parser.utils import NormalizedNode, normalize_nodes
+from xrpl.utils.txn_parser.utils.parser import get_value, group_by_account
+from xrpl.utils.txn_parser.utils.types import (
+    AccountOfferChange,
+    AccountOfferChanges,
+    CurrencyAmount,
+    OfferChange,
+)
+from xrpl.utils.xrp_conversions import drops_to_xrp
+
+LSF_SELL = 0x00020000
+
+
+def _get_offer_status(
+    node: NormalizedNode,
+) -> Literal["created", "partially-filled", "filled", "cancelled"]:
+    node_type = node["NodeType"]
+    if node_type == "CreatedNode":
+        return "created"
+    elif node_type == "ModifiedNode":
+        return "partially-filled"
+    else:  # node_type == "DeletedNode"
+        previous_fields = node.get("PreviousFields")
+        # a filled offer has previous fields
+        if previous_fields is not None:
+            return "filled"
+        # a cancelled offer has no previous fields
+        return "cancelled"
+
+
+def _derive_currency_amount(
+    currency_amount: Union[str, Dict[str, str]]
+) -> CurrencyAmount:
+    if isinstance(currency_amount, str):
+        return CurrencyAmount(currency="XRP", value=str(drops_to_xrp(currency_amount)))
+    else:
+        return CurrencyAmount(
+            currency=currency_amount["currency"],
+            issuer=currency_amount["issuer"],
+            value=currency_amount["value"],
+        )
+
+
+def _calculate_delta(
+    final_amount: CurrencyAmount,
+    previous_amount: CurrencyAmount,
+) -> str:
+    final_value = get_value(final_amount)
+    previous_value = get_value(previous_amount)
+    delta = final_value - previous_value
+    return str(delta)
+
+
+def _get_change_amount(
+    node: NormalizedNode,
+    status: Literal["created", "partially-filled", "filled", "cancelled"],
```

Comment:
You don't actually need the `status` parameter here since we know that the status is "cancelled" if `final_fields` exists (which we check anyway), and we know that the status is "created" if `new_fields` exists (which we also check). Then the other two possible statuses are never checked. 

By doing that, you can bring the if-statements one level down :)